Core/clim-core/panes.lisp

```diff
@@ -363,11 +363,38 @@ order to produce a double-click")
   (print-unreadable-object (pane sink :type t :identity t)
     (prin1 (pane-name pane) sink)))
 
+(defgeneric find-concrete-pane-class (pane-realizer pane-type)
+  (:documentation "Resolves abstract pane name to a concrete pane ~
+class which may be instatinated. Function specializes on its first ~
+argument. Returns NIL if no concrete class exists.")
+  (:method ((realizer t) pane-type)
+    ;; Default method tries to resolve the pane name as specified by a
+    ;; convention mentioned in the spec. Function is little
+    ;; complicated because we preserve old semantics adding rules to
+    ;; the class name resolution. Resolution works as follows:
+    ;;
+    ;; 1. Abstract mapping always takes a priority. When it exists we
+    ;;    expect that class is defined and signal error otherwise.
+    ;; 2. When the symbol is in clim/climi/keyword package:
+    ;;    - look for a class `CLIMI::{symbol-name}-PANE'
+    ;;    - look for a class `CLIMI::{symbol-name}'
+    ;; 3. Otherwise find class named by the symbol.
+    (when-let ((mapped (get pane-type 'concrete-pane-class-name)))
+      (return-from find-concrete-pane-class (find-class mapped t)))
+    (if (let ((symbol-package (symbol-package pane-type)))
+          (or (eql symbol-package (find-package '#:clim))
+              (eql symbol-package (find-package '#:climi))
+              (eql symbol-package (find-package '#:keyword))))
+        (let* ((symbol-name (symbol-name pane-type))
+               (clim-symbol (find-symbol symbol-name '#:climi)))
+          (if-let ((mapped (get clim-symbol 'concrete-pane-class-name)))
+            (find-class mapped t)
+            (let ((proper-name (concatenate 'string symbol-name "-PANE")))
+              (or (find-class (find-symbol proper-name '#:climi) nil)
+                  (find-class (find-symbol symbol-name '#:climi) nil))))            )
```

Comment:
Right paren preceded by bogus whitespace.

Rather than relying on `(find-class nil nil)` returning `nil` (it could signal an error since `nil` is not a valid class name), please do something like

```lisp
(when-let ((symbol (find-symbol proper-name '#:climi)))
  (find-class symbol nil))
```

Reply:
fixed (in slightly different style, but preserving the spirit of this remark)